translate guide/why.md

harapeko opened the pull request:
これから翻訳を開始します
遅くて来週後半を予想しています

ref #28

harapeko commented:
進捗状況：ファイルを一通り翻訳→リファクタリングまで完了！
今後の流れ：清書して1コミットにまとめる→少し寝かせて微修正します

harapeko commented:
読み直し、声出しをして調整中です
そのほか、漏れがないかなど。

それがおわったら1コミットにまとめてレビューを出す予定です

harapeko committed:
translate guide/why.md

harapeko commented on guide/why.md:
```diff
@@ -1,55 +1,55 @@
+バンドラーの開発サーバーではメモリ上でバンドルを実行するものがあります。それはファイルが変更されたとき、モジュールグラフの一部を無効にするだけですが、バンドル全体を再構築して、ウェブページをリロードしなければなりません。バンドルの再構築にはコストがかかりますし、ページリロードされるとアプリケーションの現在状態は消えてしまいます。そのため、幾つかのバンドラーは HMR(ホットモジュールリプレースメント)をサポートしています：これにより、ページの変更に関係のない部分には影響を与えることなく、モジュールを「ホットリプレース」することができます。これは開発者体験を大きく改善します。- しかしながら、実際には HMR でもアプリケーションが大きくなるつれ更新速度が著しく悪化することは分かっていました。
 
-In Vite, HMR is performed over native ESM. When a file is edited, Vite only needs to precisely invalidate the chain between the edited module and its closest HMR boundary (most of the time only the module itself), making HMR updates consistently fast regardless of the size of your application.
+Vite では、HMR をネイティブ ESM 上で 行います。ファイルが編集されたとき、Vite は編集されたモジュールと最も近い HMR boundary 間のチェーンを厳密に無効化することで(大抵はモジュール本体だけです)、HMR による更新はアプリケーションのサイズに関係なく一貫して高速で実行されます。
```
HMR boundaryは訳しようがなかったので、そのままにしています。
また、これの概念はkazuponさんのスライドが理解に役立ちました
https://speakerdeck.com/kazupon/native-esm-powered-web-dev-build-tool?slide=54

takurinton replied:
なるほど、良さそうです！！

harapeko commented:
@takurinton 
レビューおねがいします！🙇🏻‍♂️

作業ファイルに関してはtextlintをパスしてあります👍

harapeko committed:
fix guide/whyの長音削除した

harapeko committed:
fix guide/whyの「:」を「: 」に戻した

why?
ガイドラインが原文に合わせていたので

harapeko commented:
ガイドラインに合わせる修正を2点コミットしました

harapeko commented on guide/why.md:
```diff
@@ -1,55 +1,55 @@
+バンドラベースのビルドセットアップでファイルが編集されたとき、全てのバンドルを再構築することが非効率なことは明白です: 更新スピードはアプリケーションのサイズに応じて線形的に低下します。
 
-Some bundler dev server runs the bundling in memory so that it only needs to invalidate part of its module graph when a file changes, but it still needs to re-construct the entire bundle and reload the web page. Reconstructing the bundle can be expensive, and reloading the page blows away the current state of the application. This is why some bundlers support Hot Module Replacement (HMR): allowin ...
+バンドラの開発サーバではメモリ上でバンドルを実行するものがあります。それはファイルが変更されたとき、モジュールグラフの一部を無効にするだけですが、バンドル全体を再構築して、ウェブページをリロードしなければなりません。バンドルの再構築にはコストがかかりますし、ページリロードされるとアプリケーションの現在状態は消えてしまいます。そのため、幾つかのバンドラは HMR(ホットモジュールリプレースメント)をサポートしています: これにより、ページの変更に関係のない部分には影響を与えることなく、モジュールを「ホットリプレース」することができます。これは開発者体験を大きく改善します。- しかしながら、実際には HMR でもアプリケーションが大きくなるつれ更新速度が著しく悪化することは分かっていました。
```
>分かっていました

ここのfoundをどうするか悩みました。
誰が、いつの時点で、更新速度の悪化に気がついていたかですよね。

誰が？：ここでは、HMRを作った人たちというよりかは、我々使用者が使っていたケースな気がします。
いつの時点で？：高速だと思って使っていたら、おや？遅くなるぞ？と、後から気がついた様子

なので、ここは「分かってきました」、「見つかりました」、「発見されました」が適切な気がしてきました

takurinton replied:
@harapeko 
おはようございます。遅くなり申し訳ありません。
ここに関しては
「見つかりました」
が良さそうですね

takurinton commented on guide/why.md:
```diff
@@ -1,55 +1,55 @@
+ES モジュールがブラウザで利用できるようになるまで、開発者はモジュール化された JavaScript を生成するネイティブの仕組みを持っていませんでした。これは、私たちが「バンドル」のコンセプトに慣れ親しんでいる理由でもあります: すなわち、ブラウザで実行可能なようにソースモジュールをクロール、処理し、連結するツールを使用しています。
 
-Over time we have seen tools like [webpack](https://webpack.js.org/), [Rollup](https://rollupjs.org) and [Parcel](https://parceljs.org/), which greatly improved the development experience for frontend developers.
+時を経て [webpack](https://webpack.js.org/)、[Rollup](https://rollupjs.org)、[Parcel](https://parceljs.org/) のようなツールが登場し、フロントエンド開発者の開発体験は大きく向上されました。
```
> 時を経て [webpack](https://webpack.js.org/) や [Rollup](https://rollupjs.org)、[Parcel](https://parceljs.org/) のようなツールが登場し、フロントエンド開発者の開発体験は大きく向上されました。

とした方が自然かもしれません。

takurinton commented on guide/why.md:
```diff
@@ -1,55 +1,55 @@
+Vite では、HMR をネイティブ ESM 上で 行います。ファイルが編集されたとき、Vite は編集されたモジュールと最も近い HMR boundary 間のチェーンを厳密に無効化することで(大抵はモジュール本体だけです)、HMR による更新はアプリケーションのサイズに関係なく一貫して高速で実行されます。
 
-Vite also leverages HTTP headers to speed up full page reloads (again, let the browser do more work for us): source code module requests are made conditional via `304 Not Modified`, and dependency module requests are strongly cached via `Cache-Control: max-age=31536000,immutable` so they don't hit the server again once cached.
+また、Vite は HTTP ヘッダーを活用して、フルページのリロードも高速化します (ここでも、ブラウザにはもっと働いてもらいます): ソースコードモジュールのリクエストでは `304 Not Modified` を利用して条件が作成されます。そして、依存モジュールのリクエストでは、一度キャッシュされたものが再びサーバにヒットしないよう、`Cache-Control: max-age=31536000,immutable` を利用して強力にキャッシュされます。
```
`強力にキャッシュされる` という表現についてですが、ここは `積極的にキャッシュされる` にしたほうが良さそうです。

以前、[Featureページの翻訳 #10](https://github.com/vuejs-jp/vite-docs-ja/pull/10#discussion_r587910019) で同じような話がありました！

takurinton reviewed (commented):
@harapeko 
翻訳ありがとうございます。
レビューしました！
1つ目のコメントに関しては提案のようなものなのでどちらにするかはお任せします！
`強力にキャッシュされる` の点だけ目を通していただけると嬉しいです！よろしくお願いします！

harapeko committed:
update 分かっていました to 分かってきました

why?
分かっていましただと、
HMRを組み込んだ実装者か、
すごく精通している開発者になると思いますが、
ここでは前述の一般的な開発者か、
遅くなることはないと思っていたHMRを組み込んだ実装者だと思うので。

harapeko committed:
update 「A、B、C のような〜」 を 「A や B、C のような」に変更して自然な形にした

how?
webpack、Rollup、Parcel のような を
webpack や Rollup、Parcel のような にした

提案元
https://github.com/vuejs-jp/vite-docs-ja/pull/36/files#discussion_r606028724

harapeko committed:
update 強力にキャッシュされる to 積極的にキャッシュされる

why?
文言統一のため

経緯としては、
ここでのStronglyは
Viteが仕組み上、依存関係を強くキャッシュするが
手動編集して無効にしたり、デバッグしたりする方法もある。

すべてではないという意図、経緯のため。

このコメントが誤っている可能性もあるので、
違和感を感じたらIssueを探ってみてください

一旦は文言統一。

ref
・提案元：https://github.com/vuejs-jp/vite-docs-ja/pull/36/files#discussion_r606030876
・経緯：https://github.com/vuejs-jp/vite-docs-ja/pull/10#discussion_r587910019
・キャッシングについて：https://github.com/takurinton/vite-docs-ja/blob/featurepage-translate/guide/dep-pre-bundling.md#browser-cache

harapeko commented:
@takurinton 再レビューおねがいします！
ご提案いただいた箇所を取り込まさせていただきました！
(レビューありがとうございます！)

takurinton commented:
@harapeko 
素早い対応ありがとうございます！
確認しました！良さそうなのでマージします！
またよろしくお願いします！🙇‍♂️
